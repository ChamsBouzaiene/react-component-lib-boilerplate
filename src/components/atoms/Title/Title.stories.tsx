import React from 'react';
import { Story, Meta } from '@storybook/react';

import Title, { Props } from './Title';

export default {
    title: 'Atoms/Title',
    component: Title,
} as Meta;

const Template: Story<Props> = (args) => <Title {...args} />;

export const Default = Template.bind({});
Default.args = {
    children: 'Welcome',
};

export const small = Template.bind({});
small.args = {
    children: 'Welcome',
    fontSize: 25,
};

export const Colorful = Template.bind({});
Colorful.args = {
    children: 'Welcome',
    color: 'tomato',
};

export const MonoText = Template.bind({});
MonoText.args = {
    children: 'Welcome',
    color: 'tomato',
    fontFamily: 'monospace',
};
