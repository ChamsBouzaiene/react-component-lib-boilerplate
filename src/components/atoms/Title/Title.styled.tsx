import React from 'react';
import styled from 'styled-components';
import {
    typography,
    TypographyProps,
    color,
    ColorProps,
    space,
    fontSize,
    SpaceProps,
    FontSizeProps,
} from 'styled-system';

interface Props extends ColorProps, SpaceProps, FontSizeProps, TypographyProps {
    children: React.ReactNode;
    verticalAlign?: string;
}

export const StyledTitle = styled('h1')<Props>(
    {
        display: 'inline-block',
    },
    color,
    space,
    fontSize,
    typography,
);
