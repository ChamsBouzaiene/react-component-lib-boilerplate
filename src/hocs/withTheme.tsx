import React from 'react';
import { ThemeProvider } from 'styled-components';
import darkTheme from '../theme';

export function withTheme<P>(WrappedComponent: React.ComponentType<P>): React.FC<P> {
    const ComponentWithTheme = (props: P) => {
        return (
            <ThemeProvider theme={darkTheme}>
                <WrappedComponent {...props} />
            </ThemeProvider>
        );
    };
    return ComponentWithTheme;
}
export default withTheme;
