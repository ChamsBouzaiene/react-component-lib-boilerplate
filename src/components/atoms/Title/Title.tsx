import React, { HTMLAttributes } from 'react';
import PropTypes from 'prop-types';
import { StyledTitle } from './Title.styled';
import { TypographyProps, SpaceProps, FontSizeProps } from 'styled-system';

export interface Props extends HTMLAttributes<HTMLElement>, SpaceProps, FontSizeProps, TypographyProps {
    children: React.ReactNode;
    verticalAlign?: string;
}

const Title: React.FC<Props> = ({ children, ...props }) => <StyledTitle {...props}>{children}</StyledTitle>;

Title.propTypes = {
    children: PropTypes.node.isRequired,
};

export default Title;
