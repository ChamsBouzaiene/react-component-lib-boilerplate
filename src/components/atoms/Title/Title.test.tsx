import React from 'react';
import { render, screen } from '@testing-library/react';
import Title from './Title';
import { withTheme } from '../../../hocs/withTheme';

const ThemedTitle = withTheme(Title);

test('renders a default Title', () => {
    const { baseElement } = render(<ThemedTitle>TESTING</ThemedTitle>);
    expect(screen.getByText('TESTING')).toBeInTheDocument();
    expect(baseElement).toMatchSnapshot();
});
