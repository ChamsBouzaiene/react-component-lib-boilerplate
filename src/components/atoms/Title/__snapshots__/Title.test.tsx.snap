// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a default Title 1`] = `
<body>
  <div>
    <h1
      class="sc-bdnxRM icaPHF"
    >
      TESTING
    </h1>
  </div>
</body>
`;
